### PR TITLE
Allow replica identity on compressed hypertables

### PR DESCRIPTION
**Pull request by JamesGuthrie**

### CODE REVIEW

**Review by svenklemm — APPROVED** [review]

File: tsl/test/expected/compression_ddl.out [hunk 1]
```diff
@@ -54,6 +54,48 @@ DROP INDEX new_index;
 ALTER TABLE test1 SET (fillfactor=100);
 ALTER TABLE test1 RESET (fillfactor);
 ALTER TABLE test1 ALTER COLUMN b SET STATISTICS 10;
+-- ensure that REPLICA IDENTITY works
+ALTER TABLE test1 REPLICA IDENTITY FULL;
+SELECT relname, relreplident FROM pg_class WHERE relname = 'test1' ORDER BY relname;
+ relname | relreplident 
+---------+--------------
+ test1   | f
```
Comment (fabriziomello) [inline tsl/test/expected/compression_ddl.out:62]: Can u run it before and after changing the replica identity to make evident that it changed properly.

File: tsl/test/expected/compression_ddl.out [hunk 2]
```diff
@@ -54,6 +54,48 @@ DROP INDEX new_index;
...
+(1 row)
+
+-- the chunk's setting should also change to FULL
+SELECT relname, relreplident FROM show_chunks('test1') ch INNER JOIN pg_class c ON (ch = c.oid) ORDER BY relname;
```
Comment (fabriziomello) [inline tsl/test/expected/compression_ddl.out:66]:
```suggestion
SELECT relreplident, COUNT(*) FROM show_chunks('test1') ch INNER JOIN pg_class c ON (ch = c.oid) GROUP BY 1 ORDER BY 1;
```
Decrease the test output size... also be good to run it before and after change the replica identity.

File: tsl/test/expected/compression_ddl.out (hunk 1 above)
Comment (JamesGuthrie) [reply tsl/test/expected/compression_ddl.out:68]: done

File: tsl/test/expected/compression_ddl.out (hunk 2 above)
Comment (JamesGuthrie) [reply tsl/test/expected/compression_ddl.out]: done

**Review by fabriziomello — APPROVED** [review]

File: tsl/test/expected/compression_ddl.out (hunk 1 above)
Comment (JamesGuthrie) [reply tsl/test/expected/compression_ddl.out:74]: done

**Pull request merged**